test/common/upstream/load_balancer_fuzz_base.cc

```diff
@@ -0,0 +1,212 @@
+    // Health flags are not currently directly used by most load balancers, but
+    // they may be added and also are used by other components.
+    // There are three health flags that map to Host::Health::Degraded, FAILED_ACTIVE_HC,
+    // FAILED_OUTLIER_CHECK, and FAILED_EDS_HEALTH. Choose one hardcoded for simplicity.
+    host_set.hosts_[index]->healthFlagSet(Host::HealthFlag::FAILED_ACTIVE_HC);
+  }
+
+  ENVOY_LOG_MISC(trace, "Hosts made excluded at priority level {}: {}", priority_of_host_set,
+                 absl::StrJoin(subsets.at(2), " "));
+
+  // Handle updating health flags for hosts_per_locality_
+
+  // The index within the array of the vector represents the locality
+  std::array<HostVector, 3> healthy_hosts_per_locality;
+  std::array<HostVector, 3> degraded_hosts_per_locality;
+  std::array<HostVector, 3> excluded_hosts_per_locality;
+
+  // Wrap those three in an array here, where the index represents health status of
+  // healthy/degraded/excluded, used for indexing during iteration through subsets
+  std::array<std::array<HostVector, 3>, 3> locality_health_statuses = {
+      healthy_hosts_per_locality, degraded_hosts_per_locality, excluded_hosts_per_locality};
+
+  // Iterate through subsets
+  for (uint8_t health_status = 0; health_status < locality_health_statuses.size();
+       health_status++) {
+    for (uint8_t index : subsets.at(health_status)) { // Each subset logically represents a health
+                                                      // status
+      // If the host is in a locality, we have to update the corresponding health status host vector
+      if (!(locality_indexes_.find(index) == locality_indexes_.end())) {
+        // After computing the host index subsets, we want to propagate these changes to a host set
+        // by building and using these host vectors
+        uint8_t locality = locality_indexes_[index];
+        locality_health_statuses[health_status][locality].push_back(host_set.hosts_[index]);
+        ENVOY_LOG_MISC(trace, "Added host at index {} in locality {} to health status {}", index,
+                       locality_indexes_[index] + 1, health_status + 1);
+      }
+    }
+  }
+
+  // This overrides what is currently present in the host set, thus not having to explicitly call
+  // vector.clear()
+  host_set.healthy_hosts_per_locality_ =
+      makeHostsPerLocality({healthy_hosts_per_locality[0], healthy_hosts_per_locality[1],
+                            healthy_hosts_per_locality[2]});
+  host_set.degraded_hosts_per_locality_ =
+      makeHostsPerLocality({degraded_hosts_per_locality[0], degraded_hosts_per_locality[1],
+                            degraded_hosts_per_locality[2]});
+  host_set.excluded_hosts_per_locality_ =
+      makeHostsPerLocality({excluded_hosts_per_locality[0], excluded_hosts_per_locality[1],
+                            excluded_hosts_per_locality[2]});
+
+  host_set.runCallbacks({}, {});
```

Comment:
Did you confirm this is wired so that we observe load balancer rebuilds when we hit this statement? Maybe add a comment on this.

Reply:
Sorry - what does "this is wired" mean?

Reply:
I mean, do the callbacks actually do anything? Are there registered callbacks that rebuild load balancing data structures on health change?

Reply:
Took a look around - yeah, they do through these cb_helpers_ mocked over in Mock Priority Set: https://github.com/envoyproxy/envoy/blob/8b9f68848877502fd32fc85c281c442310963494/test/mocks/upstream/priority_set.cc#L46. A callback which rebuilds data structures in passed into that here in the source code: https://github.com/envoyproxy/envoy/blob/5875f23b60aeeb49e038596d8de36446c6728f5d/source/common/upstream/load_balancer_impl.cc#L116. Will add comment explaining.